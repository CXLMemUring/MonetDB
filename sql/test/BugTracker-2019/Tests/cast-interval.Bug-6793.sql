CREATE TABLE INTERVAL_TBL (f1 interval second);
INSERT INTO INTERVAL_TBL (f1) VALUES (interval '1' second);
INSERT INTO INTERVAL_TBL (f1) VALUES (interval '2' second);
INSERT INTO INTERVAL_TBL (f1) VALUES (interval '3' second);
INSERT INTO INTERVAL_TBL (f1) VALUES (interval '4' second);
SELECT f1, cast(f1 as int), cast(f1 as dec(8,3)), cast(f1 as real) FROM INTERVAL_TBL; -- error
SELECT count(f1), sum(f1), sum(f1), sum(f1), sum(f1) FROM INTERVAL_TBL;
DROP TABLE INTERVAL_TBL;

CREATE TABLE INTERVAL_TBL (f1 interval month);
INSERT INTO INTERVAL_TBL (f1) VALUES (interval '1' month);
INSERT INTO INTERVAL_TBL (f1) VALUES (interval '2' month);
INSERT INTO INTERVAL_TBL (f1) VALUES (interval '3' month);
INSERT INTO INTERVAL_TBL (f1) VALUES (interval '4' month);
SELECT f1, cast(f1 as int) FROM INTERVAL_TBL; -- error
SELECT f1, cast(f1 as dec(3,0)) FROM INTERVAL_TBL;  -- returns error: types month_interval(3,0) and decimal(3,0) are not equal for column 'f1'
SELECT f1, cast(f1 as real) FROM INTERVAL_TBL;  --  returns error: types month_interval(3,0) and real(24,0) are not equal for column 'f1'
SELECT count(f1), sum(f1), sum(f1), sum(f1), sum(f1) FROM INTERVAL_TBL;
DROP TABLE INTERVAL_TBL;

