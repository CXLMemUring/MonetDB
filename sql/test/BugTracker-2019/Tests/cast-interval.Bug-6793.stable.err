stderr of test 'cast-interval.Bug-6793` in directory 'sql/test/BugTracker-2019` itself:


# 13:25:35 >  
# 13:25:35 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=33951" "--set" "mapi_usock=/var/tmp/mtest-5984/.s.monetdb.33951" "--forcemito" "--dbpath=/home/dinther/dev/dev/INSTALL/var/MonetDB/mTests_sql_test_BugTracker-2019" "--set" "embedded_c=true"
# 13:25:35 >  

MAPI  = (monetdb) /var/tmp/mtest-440686/.s.monetdb.35626
QUERY = SELECT f1, cast(f1 as int), cast(f1 as dec(8,3)), cast(f1 as real) FROM INTERVAL_TBL; -- error
ERROR = !types sec_interval(13,0) and int(32,0) are not equal for column 'f1'
CODE  = 42000
MAPI  = (monetdb) /var/tmp/mtest-440686/.s.monetdb.35626
QUERY = SELECT f1, cast(f1 as int) FROM INTERVAL_TBL; -- error
ERROR = !types month_interval(3,0) and int(32,0) are not equal for column 'f1'
CODE  = 42000
MAPI  = (monetdb) /var/tmp/mtest-440686/.s.monetdb.35626
QUERY = SELECT f1, cast(f1 as dec(3,0)) FROM INTERVAL_TBL;  -- returns error: types month_interval(3,0) and decimal(3,0) are not equal for column 'f1'
ERROR = !types month_interval(3,0) and decimal(3,0) are not equal for column 'f1'
CODE  = 42000
MAPI  = (monetdb) /var/tmp/mtest-440686/.s.monetdb.35626
QUERY = SELECT f1, cast(f1 as real) FROM INTERVAL_TBL;  --  returns error: types month_interval(3,0) and real(24,0) are not equal for column 'f1'
ERROR = !types month_interval(3,0) and real(24,0) are not equal for column 'f1'
CODE  = 42000

# 13:25:36 >  
# 13:25:36 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-5984" "--port=33951"
# 13:25:36 >  


# 13:25:36 >  
# 13:25:36 >  "Done."
# 13:25:36 >  

