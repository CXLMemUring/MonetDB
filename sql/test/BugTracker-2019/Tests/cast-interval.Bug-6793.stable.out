stdout of test 'cast-interval.Bug-6793` in directory 'sql/test/BugTracker-2019` itself:


# 13:25:35 >  
# 13:25:35 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=33951" "--set" "mapi_usock=/var/tmp/mtest-5984/.s.monetdb.33951" "--forcemito" "--dbpath=/home/dinther/dev/dev/INSTALL/var/MonetDB/mTests_sql_test_BugTracker-2019" "--set" "embedded_c=true"
# 13:25:35 >  

# MonetDB 5 server v11.35.4 (hg id: e54cc7e9164a)
# This is an unreleased version
# Serving database 'mTests_sql_test_BugTracker-2019', using 8 threads
# Compiled for x86_64-pc-linux-gnu/64bit with 128bit integers
# Found 31.243 GiB available main-memory of which we use 25.463 GiB
# Copyright (c) 1993 - July 2008 CWI.
# Copyright (c) August 2008 - 2019 MonetDB B.V., all rights reserved
# Visit https://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://localhost.localdomain:33951/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-5984/.s.monetdb.33951
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded

# 13:25:36 >  
# 13:25:36 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-5984" "--port=33951"
# 13:25:36 >  

#CREATE TABLE INTERVAL_TBL (f1 interval second);
#INSERT INTO INTERVAL_TBL (f1) VALUES (1.123);
[ 1	]
#INSERT INTO INTERVAL_TBL (f1) VALUES (2.123);
[ 1	]
#INSERT INTO INTERVAL_TBL (f1) VALUES (3.123);
[ 1	]
#INSERT INTO INTERVAL_TBL (f1) VALUES (4.123);
[ 1	]
#SELECT f1, cast(f1 as int), cast(f1 as dec(8,3)), cast(f1 as real) FROM INTERVAL_TBL;
% sys.interval_tbl,	sys.L1,	sys.L2,	sys.L3 # table_name
% f1,	L1,	L2,	L3 # name
% sec_interval,	int,	decimal,	real # type
% 5,	1,	10,	15 # length
[ 1.123,	1,	1.123,	1.123	]
[ 2.123,	2,	2.123,	2.123	]
[ 3.123,	3,	3.123,	3.123	]
[ 4.123,	4,	4.123,	4.123	]
#SELECT count(f1), sum(f1), cast(sum(f1) as int), cast(sum(f1) as dec(8,3)), cast(sum(f1) as real) FROM INTERVAL_TBL;
% sys.L2,	sys.L3,	sys.L5,	sys.L7,	sys.L11 # table_name
% L2,	L3,	L5,	L7,	L11 # name
% bigint,	sec_interval,	int,	decimal,	real # type
% 1,	6,	2,	10,	15 # length
[ 4,	10.492,	10,	10.492,	10.492	]
#DROP TABLE INTERVAL_TBL;
#CREATE TABLE INTERVAL_TBL (f1 interval month);
#INSERT INTO INTERVAL_TBL (f1) VALUES (1);
[ 1	]
#INSERT INTO INTERVAL_TBL (f1) VALUES (2);
[ 1	]
#INSERT INTO INTERVAL_TBL (f1) VALUES (3);
[ 1	]
#INSERT INTO INTERVAL_TBL (f1) VALUES (4);
[ 1	]
#SELECT f1, cast(f1 as int) FROM INTERVAL_TBL;
% sys.interval_tbl,	sys.L1 # table_name
% f1,	L1 # name
% month_interval,	int # type
% 1,	1 # length
[ 1,	1	]
[ 2,	2	]
[ 3,	3	]
[ 4,	4	]
#SELECT f1, cast(f1 as dec(3,0)) FROM INTERVAL_TBL;
% sys.interval_tbl,	sys.L1 # table_name
% f1,	L1 # name
% month_interval,	decimal # type
% 1,	5 # length
[ 1,	1	]
[ 2,	2	]
[ 3,	3	]
[ 4,	4	]
#SELECT f1, cast(f1 as real) FROM INTERVAL_TBL;
% sys.interval_tbl,	sys.L1 # table_name
% f1,	L1 # name
% month_interval,	real # type
% 1,	15 # length
[ 1,	1	]
[ 2,	2	]
[ 3,	3	]
[ 4,	4	]
#SELECT count(f1), sum(f1), cast(sum(f1) as int), cast(sum(f1) as dec(8,3)), cast(sum(f1) as real) FROM INTERVAL_TBL;
% sys.L2,	sys.L3,	sys.L5,	sys.L7,	sys.L11 # table_name
% L2,	L3,	L5,	L7,	L11 # name
% bigint,	month_interval,	int,	decimal,	real # type
% 1,	2,	2,	10,	15 # length
[ 4,	10,	10,	10.000,	10.000	]
#DROP TABLE INTERVAL_TBL;

# 13:25:36 >  
# 13:25:36 >  "Done."
# 13:25:36 >  

